tests/e2e/preferential_topology.go

```diff
@@ -781,4 +781,993 @@ var _ = ginkgo.Describe("[Preferential-Topology] Preferential-Topology-Provision
 		verifyPVnodeAffinityAndPODnodedetailsForStatefulsetsLevel5(ctx, client, statefulset,
 			namespace, allowedTopologyForRack2, false)
 	})
+
+	/*
+		Testcase-6:
+			Multiple tags are assigned to Shared datastore with multiple allowed topology
+
+			Steps
+			1. Assign Tag rack-1, Category "cns.vmware.topology-preferred-datastores" to datastore
+			accessible across all racks (ex - sharedVMFS-12)
+			2. Assign Tag rack-2, Category "cns.vmware.topology-preferred-datastores" to datastore
+			accessible across all racks (ex - sharedVMFS-12)
+			3. Assign Tag rack-3, Category "cns.vmware.topology-preferred-datastores" to datastore
+			accessible across all racks (ex - sharedVMFS-12)
+			4. Create SC with WFC binding mode and allowed topologies set to rack-1/rack-2/rack-3 in the SC
+			5. Create StatefulSet with replica 3 with the above SC
+			6. Wait for all the StatefulSet to come up
+			7. Wait for PV , PVC to reach bound and POD to reach running state
+			8. Describe PV and verify node affinity details should contain allowed topology details.
+			9. Verify volume provisioning should be on the preferred datastore.
+			10. Make sure POD is running on the same node as mentioned in node affinity details
+			11. Perform cleanup. Delete StatefulSet, PVC, PV and SC.
+			12. Remove datastore preference tags as part of cleanup.
+	*/
+	ginkgo.It("Assign multiple tags to preferred shared datastore which is shared across all "+
+		"racks with multiple allowed topologies", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		preferredDatastoreChosen = 1
+		preferredDatastorePaths = nil
+
+		// choose preferred datastore
+		ginkgo.By("Tag preferred datatsore for volume provisioning")
+		preferredDatastorePaths, err = tagPreferredDatastore(masterIp, allowedTopologyRacks[0],
+			preferredDatastoreChosen, shareddatastoreListMap, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = tagSameDatastoreAsPreferenceToDifferentRacks(masterIp, allowedTopologyRacks[1],
+			preferredDatastoreChosen, preferredDatastorePaths)
```

Comment:
tagSameDatastoreAsPreferenceToDifferentRacks  -> Can we put this in a loop?

Reply:
code corrected